sky/spot/spot_utils.py

```diff
@@ -199,22 +200,26 @@ def cancel_job_by_name(job_name: str) -> str:
 def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
     """Stream logs by job id."""
     controller_status = job_lib.get_status(job_id)
-    while (controller_status != job_lib.JobStatus.RUNNING and
-           (controller_status is None or not controller_status.is_terminal())):
-        status_str = 'None'
-        if controller_status is not None:
-            status_str = controller_status.value
-        logger.info(
-            'Waiting for the spot controller process to be RUNNING (status: '
-            f'{status_str}).')
-        time.sleep(_LOG_STREAM_CHECK_CONTROLLER_GAP_SECONDS)
-        controller_status = job_lib.get_status(job_id)
-
-    job_status = spot_state.get_status(job_id)
-    while job_status is None:
-        logger.info('Waiting for the spot job to be started.')
-        time.sleep(1)
+    status_msg = ('[bold cyan]Waiting for controller, controller status: {}.[/]'
+                  'It may take a few minutes.')
+    status_display = rich.status.Status('[bold cyan]Waiting for logs...[/]')
```

Comment:
Q: I can't seem to trigger this log with `sky spot launch sleep 10000`.

Reply:
It is the general initial status to avoid empty logging and will be soon updated later. I changed it to `Waiting for controller process to be RUNNING`, so it is consistent with the first logging afterwards.